cmd/utils.go

```diff
@@ -178,37 +176,46 @@ type prefixSSEPair struct {
 
 // parse list of comma separated alias/prefix=sse key values entered on command line and
 // construct a map of alias to prefix and sse pairs.
-func parseEncryptionKeys(ssekeys string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
+func parseEncryptionKeys(sseKeys string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
 	encMap = make(map[string][]prefixSSEPair)
-	if ssekeys == "" {
+	if sseKeys == "" {
 		return
 	}
-	fields := strings.Split(ssekeys, ",")
-	r := regexp.MustCompile("([^=\\s]+)=([^\\s]+)")
-	for _, field := range fields {
-		field = strings.TrimSpace(field)
-		matches := r.FindStringSubmatch(field)
-		if len(matches) != 3 {
-			return nil, probe.NewError(errors.New("sse-c prefix should be of the form prefix=key"))
+	prefix := ""
+	ssekey := ""
+	index := 0 // start index of prefix
+	vs := 0    // start index of sse-c key
+	sseKeyLen := 32
+	delim := 1
+	l := len(sseKeys)
+	for index < l {
+		e := strings.Index(sseKeys[index:], "=")
+		if e == -1 {
+			return nil, probe.NewError(errors.New("sse-c prefix should be of the form prefix1=key1,... "))
 		}
-		key := matches[1]
-		value := matches[2]
-		alias, _ := url2Alias(key)
-		if len(value) != 32 {
+		prefix = sseKeys[index : index+e]
+		// validate alias
+		alias, _, hostCfg, err := expandAlias(prefix)
+		if hostCfg == nil || err != nil {
+			return nil, probe.NewError(errors.New("sse-c prefix " + prefix + " has invalid alias"))
+		}
+		vs = e + 1 + index
+		if vs+32 > l {
 			return nil, probe.NewError(errors.New("sse-c key should be 32 bytes long"))
 		}
-		prefix := strings.TrimSpace(key)
+		ssekey = sseKeys[vs : vs+sseKeyLen]
+		if (vs+sseKeyLen < l) && sseKeys[vs+sseKeyLen] != ',' {
+			return nil, probe.NewError(errors.New("sse-c prefix=secret should be delimited by , and secret should be 32 bytes long"))
+		}
 		if _, ok := encMap[alias]; !ok {
 			encMap[alias] = make([]prefixSSEPair, 0)
 		}
-		ps := prefixSSEPair{prefix: prefix, sseKey: value}
+		ps := prefixSSEPair{prefix: prefix, sseKey: ssekey}
 		encMap[alias] = append(encMap[alias], ps)
+		// advance index 33 bytes for the next key start
```

Comment:
the comment should be,
`// advance index by (sseKeyLen + delim) bytes for the next key start index`

Reply:
done